### PR TITLE
HZN-1033: Move requisition/foreign-source to database

### PR DESCRIPTION
**Pull request by mvrueden**
* JIRA: https://issues.opennms.org/browse/HZN-1033

### CODE REVIEW

**Comment by mvrueden** [comment]
I merged with the latest changes from `develop`. The `import job` exception should no longer occur, as it was fixed in upstream `develop`.

File: features/events/traps/src/test/java/org/opennms/netmgt/trapd/TrapdConfigReloadIT.java [hunk 1]
```diff
@@ -1,107 +0,0 @@
-/*******************************************************************************
```
Comment (j-white) [inline features/events/traps/src/test/java/org/opennms/netmgt/trapd/TrapdConfigReloadIT.java]: How is this related to HZN-1033?

File: features/minion/heartbeat/consumer/src/main/java/org/opennms/minion/heartbeat/consumer/HeartbeatConsumer.java [hunk 2]
```diff
@@ -181,71 +190,66 @@ private void provision(final OnmsMinion minion,
...
+                requisitionService.saveOrUpdateRequisition(nextRequisition);
+
+                alteredForeignSources.add(nextForeignSource);
             }
+            return null;
+        });
+
+        for (final String alteredForeignSource : alteredForeignSources) {
+            requisitionService.triggerImport(
+                    new ImportRequest("Web").withForeignSource(alteredForeignSource));
```
Comment (j-white) [inline features/minion/heartbeat/consumer/src/main/java/org/opennms/minion/heartbeat/consumer/HeartbeatConsumer.java]: Web?

File: opennms-base-assembly/src/main/filtered/etc/create.sql [hunk 3]
```diff
@@ -2474,3 +2474,139 @@ CREATE TABLE topo_layout_vertex_positions (
 	CONSTRAINT fk_topo_layout_vertex_positions_vertex_position_id FOREIGN KEY (vertex_position_id)
 	REFERENCES topo_vertex_position (id) ON DELETE CASCADE
 );
+<<<<<<< HEAD
```
Comment (j-white) [inline opennms-base-assembly/src/main/filtered/etc/create.sql]: HEAD of?

File: opennms-doc/releasenotes/src/asciidoc/upgrade/upgrade-20-21.adoc [hunk 4]
```diff
@@ -0,0 +1,45 @@
+[[upgrade-19-20]]
```
Comment (j-white) [inline opennms-doc/releasenotes/src/asciidoc/upgrade/upgrade-20-21.adoc]: 19-20 -> 20-21

File: opennms-doc/releasenotes/src/asciidoc/upgrade/upgrade-20-21.adoc [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+== Upgrade OpenNMS Horizon 19 to 20
+
+=== Foreign Sources and Requisitions
+
+The Foreign Sources and Requisitions have been moved to the database.
+They must be manually migrated. In order to make the migration as smooth as possible a `foreignsource` and `requisition` osgi karaf command has been introduced.
```
Comment (j-white) [inline opennms-doc/releasenotes/src/asciidoc/upgrade/upgrade-20-21.adoc]: We should add additional instructions on how to access the Karaf shell, since users may not have accessed it before.

Comment (j-white) [inline opennms-doc/releasenotes/src/asciidoc/upgrade/upgrade-20-21.adoc]: What is the expected behavior if in import happens to triggered before the migration is completed?

File: features/events/traps/src/test/java/org/opennms/netmgt/trapd/TrapdConfigReloadIT.java (hunk 1 above)
Comment (mvrueden) [inline features/events/traps/src/test/java/org/opennms/netmgt/trapd/TrapdConfigReloadIT.java]: While working on this, the test failed a couple of times. 
I observed the same behaviour a long time ago in another branch and bumped up the "wait time" to address the issue.
As the test kept failing in this branch again, I decided to delete it.
-> It is not directly related to HZN-1033.

File: opennms-doc/releasenotes/src/asciidoc/upgrade/upgrade-20-21.adoc (hunk 4 above)
Comment (mvrueden) [inline opennms-doc/releasenotes/src/asciidoc/upgrade/upgrade-20-21.adoc]: 👍 Thanks, I must have forgotten them when merging with latest develop.

File: opennms-base-assembly/src/main/filtered/etc/create.sql (hunk 3 above)
Comment (mvrueden) [inline opennms-base-assembly/src/main/filtered/etc/create.sql]: Yeah, good point (-:

File: features/minion/heartbeat/consumer/src/main/java/org/opennms/minion/heartbeat/consumer/HeartbeatConsumer.java (hunk 2 above)
Comment (mvrueden) [inline features/minion/heartbeat/consumer/src/main/java/org/opennms/minion/heartbeat/consumer/HeartbeatConsumer.java]: Originally all Import Requests had `Web` as their source, no matter their origin. This I probably just copied and forgot to adjust. Changed it to `this`.

File: opennms-doc/releasenotes/src/asciidoc/upgrade/upgrade-20-21.adoc (hunk 5 above)
Comment (mvrueden) [inline opennms-doc/releasenotes/src/asciidoc/upgrade/upgrade-20-21.adoc]: Added instructions.
Nothing would happen, as the requisition is not yet persisted.

The migration of each requisition is made in memory.

If an import is triggered before the migration is finished, the requisition does not yet exist and nothing should happen.

However if a requisition exists in active and pending mode, then an import may already been triggered, even if not yet both requisitions have been migrated. This scenario should probably be also considered.

**Comment by mvrueden** [comment]
I don't see this beeing wrapped up any time soon or at all. Closing PR, but I will keep the branch around for a while.
